tests/utilities_test.py

```diff
@@ -21,13 +22,15 @@ def test_save_inputs_as_dat(self):
         (model, instance) = switch_mod.solve.main(
             args=["--inputs-dir", "test_dat"], return_model=True, return_instance=True
         )
-        # To do: create a temporary file name for dat_path and delete it
-        # when the test is complete.
-        dat_path = "test_dat/complete_inputs.dat"
-        utilities.save_inputs_as_dat(model, instance, save_path=dat_path)
-        reloaded_data = DataPortal(model=model)
-        reloaded_data.load(filename=dat_path)
-        compare(reloaded_data.data(), instance.DataPortal.data())
+        temp_dir = tempfile.mkdtemp()
```

Comment:
Suggestion: it's useful to use a prefix argument, e.g.
  tempfile.mkdtemp(prefix="switch_test_")
so that if files are left behind in /tmp, it's easier to tell what's leaving them behind.